### DISCUSSION TITLE
[Solved] Error message after cargo r - Failed to select a Version for env_logger

### DISCUSSION THREAD

**Discussion by dentech82** [DennisOSRM/hms-mqtt-publisher#91] [General]
Hello,

i've getting the following error message:

"
 error: failed to select a version for `env_logger`.
    ... required by package `hms-mqtt-publish v0.2.0 (/home/data/hms-mqtt-publisher)`
versions that meet the requirements `^0.11.0` are: 0.11.1, 0.11.0

the package `hms-mqtt-publish` depends on `env_logger`, with features: `anstream ` but `env_logger` does not have these features.
 It has an optional dependency with that name, but that dependency uses the "dep:" syntax in the features table, so it does not have an implicit feature with that name.

failed to select a version for `env_logger` which could resolve this conflict

"

I'm not a linux crack,

Could anybody help me?

Using: Raspberry PI 4 with Raspbian Bookworm 64 Bit

Greets Dennis

**Comment by DennisOSRM** [comment]
Interesting. Seems to work over here. What version of rust/cargo toolchain are you using? Have you tried updating to the latest one via `rustup update`?

**Comment by mario-mu** [comment]
Same here: 
`rustc 1.63.0`
`Linux  6.1.0-17-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.69-1 (2023-12-30) x86_64 GNU/Linux`

**Reply by mario-mu** [reply]
Also did not work with `rustc 1.70.0`

**Reply by mario-mu** [reply]
Setting env_logger to v0.10.2 solves the problem for me

**Reply by dentech82** [reply]
Thank you very much.

The solution was setting the env_logger to 0.10.2 and I have to update a new version of rustc. The rustc version in the debian repository was to old.

Greets

**Reply by ejannink** [reply]
Running into the same problem: how DID you set the env_logger to another version ?
I'm attempting to get things running a a docker container based on the rust image...
